src/org/traccar/protocol/NavisProtocolDecoder.java

```diff
@@ -272,41 +317,343 @@ private Object processArray(DeviceSession deviceSession, Channel channel, ByteBu
+                    break;
+                case 19:
+                    position.set(Position.KEY_BATTERY, buf.readUnsignedShortLE() * 0.001);
+                    break;
+                case 20:
+                case 21:
+                case 22:
+                case 23:
+                case 24:
+                case 25:
+                case 26:
+                case 27:
+                    position.set(Position.PREFIX_ADC + (i - 19), buf.readUnsignedShortLE());
+                    break;
+                case 28:
+                    input = buf.readUnsignedByte();
+                    position.set(Position.KEY_INPUT, input);
+                    for (int k = 0; k < 8; k++) {
+                        position.set(Position.PREFIX_IN + (k + 1), BitUtil.check(input, k));
+                    }
+                    break;
+                case 29:
+                    short input2 = buf.readUnsignedByte();
+                    position.set(Position.KEY_INPUT, (short) (BitUtil.to(input, 8) | (input2 << 8)));
+                    for (int k = 0; k < 8; k++) {
+                        position.set(Position.PREFIX_IN + (k + 9), BitUtil.check(input2, k));
+                    }
+                    break;
+                case 30:
+                    output = buf.readUnsignedByte();
+                    position.set(Position.KEY_OUTPUT, output);
+                    for (int k = 0; k < 8; k++) {
+                        position.set(Position.PREFIX_OUT + (k + 1), BitUtil.check(output, k));
+                    }
+                    break;
+                case 31:
+                    short output2 = buf.readUnsignedByte();
+                    position.set(Position.KEY_OUTPUT, (short) (BitUtil.to(output, 8) | (output2 << 8)));
+                    for (int k = 0; k < 8; k++) {
+                        position.set(Position.PREFIX_OUT + (k + 9), BitUtil.check(output2, k));
+                    }
+                    break;
+                case 36:
+                    position.set(Position.KEY_HOURS, buf.readUnsignedIntLE() * 1000);
+                    break;
+                case 44:
+                case 45:
+                case 46:
+                case 47:
+                case 48:
+                case 49:
+                case 50:
+                case 51:
+                    position.set(Position.PREFIX_TEMP + (i - 43), buf.readByte());
+                    break;
+                case 68: // CAN Speed
+                    position.set("can-speed", buf.readUnsignedByte());
+                    break;
+                // FLEX 2.0
+                case 69:
+                    int satVisible = 0;
+                    for (int k = 0; k < 8; k++) {
+                        switch (k) {
+                            case 0:
+                                int satVisibleGLONASS = buf.readUnsignedByte();
+                                position.set("sat-visible-glonass", satVisibleGLONASS);
```

Comment:
I think it's unnecessary. Just calculate simple total.

Reply:
Fixed